cmd/kubeadm/app/node/bootstrap.go

```diff
@@ -47,11 +50,49 @@ type ConnectionDetails struct {
 // to an API endpoint
 const retryTimeout = 5
 
-// EstablishMasterConnection establishes a connection with exactly one of the provided API endpoints.
+func RetrieveKubeconfigBasedOnToken(td *kubeadmapi.TokenDiscovery, clusterInfo *kubeadmapi.ClusterInfo) (*clientcmdapi.Config, error) {
+	fmt.Println("[bootstrap] Created API client to obtain unique certificate for this node, generating keys and certificate signing request")
+
+	key, err := certutil.MakeEllipticPrivateKeyPEM()
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate private key [%v]", err)
+	}
+
+	conn, err := establishMasterConnection(td, clusterInfo)
+	if err != nil {
+		return nil, fmt.Errorf("failed to establish connection to the API server[%v]", err)
+	}
+
+	cert, err := csrutil.RequestNodeCertificate(conn.CertClient.CertificateSigningRequests(), key, conn.NodeName)
```

Comment:
This still seems to me like we're doing the TLS Bootstrap with `csrutil.RequestNodeCertificate` twice, first here and then in the "real" TLS bootstrap version.

In my understanding, we just need to return the kubeconfig we're creating in `createClients` down the road. A such kubeconfig object can then be passed to the real TLSBootstrap function